From 8010902067809ad777735712ef06c8acb6d5b03a Mon Sep 17 00:00:00 2001
From: Bruno Martins <bgcngm@gmail.com>
Date: Thu, 4 Dec 2025 13:01:03 +0000
Subject: [PATCH 31/55] Revert "SystemUIProxy: Add injectPress()"

This reverts commit 0b9cb008e80971e55be9deff81c7780ff7e67030

Reason: Google added `void onKeyEvent(int keycode, int displayId)`
making this one redundant.

Change-Id: I708fbe2558b12fe10bc78591c0cd3b6dad176552
---
 .../android/systemui/shared/recents/ISystemUiProxy.aidl  | 3 ---
 .../src/com/android/systemui/LauncherProxyService.java   | 9 ---------
 2 files changed, 12 deletions(-)

diff --git a/packages/SystemUI/shared/src/com/android/systemui/shared/recents/ISystemUiProxy.aidl b/packages/SystemUI/shared/src/com/android/systemui/shared/recents/ISystemUiProxy.aidl
index 516e01c0c..ce70eed4d 100644
--- a/packages/SystemUI/shared/src/com/android/systemui/shared/recents/ISystemUiProxy.aidl
+++ b/packages/SystemUI/shared/src/com/android/systemui/shared/recents/ISystemUiProxy.aidl
@@ -114,9 +114,6 @@ interface ISystemUiProxy {
     /** Injects button long press. */
     oneway void injectLongPress(int keyCode) = 46;
 
-    /** Injects button press. */
-    oneway void injectPress(int keyCode) = 43;
-
     /** Notifies when taskbar status updated */
     oneway void notifyTaskbarStatus(boolean visible, boolean stashed) = 47;
 
diff --git a/packages/SystemUI/src/com/android/systemui/LauncherProxyService.java b/packages/SystemUI/src/com/android/systemui/LauncherProxyService.java
index ab57dfe31..c79e64096 100644
--- a/packages/SystemUI/src/com/android/systemui/LauncherProxyService.java
+++ b/packages/SystemUI/src/com/android/systemui/LauncherProxyService.java
@@ -379,15 +379,6 @@ public class LauncherProxyService implements CallbackController<LauncherProxyLis
             });
         }
 
-        @Override
-        public void injectPress(int keyCode) throws RemoteException {
-            final int displayId = mContext.getDisplayId();
-            verifyCallerAndClearCallingIdentityPostMain("pressInjected", () -> {
-                sendEvent(KeyEvent.ACTION_DOWN, keyCode, displayId);
-                sendEvent(KeyEvent.ACTION_UP, keyCode, displayId);
-            });
-        }
-
         @Override
         public void onImeSwitcherPressed() {
             // TODO(b/204901476) We're intentionally using the default display for now since
-- 
2.52.0

